docs/0004-api-spec.md

```diff
@@ -0,0 +1,175 @@
+
+### Access Roles
+
+Access roles are a mapping of a user ID to a set of permissions. The following roles are provided by default:
+
+-   `shops/owner`: This role is assigned to the user who created the shop. This role cannot be removed if a single owner remains.
+-   `shops/admin`: This role is assigned to users who have been granted admin access to the shop. Admins can manage access roles for the shop as well as view orders and manage listings.
+
+### Policies
+
+Access roles describe sets of permissions, known as policies, that the users holding that role have. Here is a list of policies:
+
+-   `shops/listings/read`: This policy grants users read access to a shop's listings. The `everyone` role has this policy by default.
+-   `shops/listings/write`: This policy grants users write access to a shop's listings.
+-   `shops/posts/read`: This policy grants users read access to a shop's posts.
+-   `shops/posts/write`: This policy grants users write access to a shop's posts.
+-   `shops/acl/read`: This policy grants users read access to the shop's ACL.
+-   `shops/acl/write`: This policy grants users write access to the shop's ACL.
+
+## Response Format
+
+All request and response bodies will be in JSON format (`application/json`). The exact format of the response will depend on the endpoint, and is specified in the endpoint documentation.
+
+### Errors
+
+If an error occurs, the response will be in the following format:
+
+```json
+{
+	"error": {
+		"code": 400,
+		"message": "Bad Request",
+		"details": "Body is missing required field 'name'"
+	}
+}
+```
+
+**Note:** The `details` field is not guaranteed to be present, but when it is, it will contain more information about the error.
+
+## Endpoints
+
+This section outlines the endpoints for the fur.ink backend. Parameters are denoted with a colon (`:`) and are required.
+
+Endpoints are typically specified using the following format:
+
+> -   `METHOD /path/to/endpoint`: Description of endpoint
+>     -   Parameter `paramName`: Description of parameter
+>     -   Argument `argumentName` (optional, default: 50): Description of a query argument
+>     -   Body: Description of the request body
+
+If an endpoint is protected, it will be specified as such.
+
+### Pagination
+
+Some endpoints have pagination enabled. Where this is referenced, you can use the following query arguments to control the pagination:
+
+> -   `limit` (optional, default: 50): The maximum number of results to return per page
+> -   `page` (optional, default: 1): The page number to return
+
+### Users
+
+-   `GET /users/:id`: Get a specific user by ID
+    -   Parameter `id`: the user ID
+-   `GET /users/:id/shops`: Get a list of shops owned by a specific user.
+    -   Parameter `id`: the user ID
+    -   This endpoint is paginated
+-   `POST /users`: Create a new user. The exact semantics of this endpoint are to be decided.
+
+#### User Orders
+
+-   `GET /users/:id/orders`: Get a list of orders for a specific user
```

Comment:
Should this not also be in the protected end points?

Reply:
Yep it should, will fix